nightfall-client/src/services/kyc.mjs

```diff
@@ -1,23 +1,38 @@
 /**
- This module creates blockchain transactions to interact with the KYC smart contract
+ This module creates blockchain transactions to interact with the Whitelist smart contract
 */
 
 import constants from '@polygon-nightfall/common-files/constants/index.mjs';
 import { waitForContract } from '@polygon-nightfall/common-files/utils/contract.mjs';
 
-const { SHIELD_CONTRACT_NAME } = constants;
+const { KYC_CONTRACT_NAME } = constants;
 
 export async function isWhitelisted(address) {
-  const shieldContractInstance = await waitForContract(SHIELD_CONTRACT_NAME);
-  return shieldContractInstance.methods.isWhitelisted(address).call();
+  const kycContractInstance = await waitForContract(KYC_CONTRACT_NAME);
+  return kycContractInstance.methods.isWhitelisted(address).call();
 }
 
 export async function addUserToWhitelist(address) {
-  const shieldContractInstance = await waitForContract(SHIELD_CONTRACT_NAME);
-  return shieldContractInstance.methods.addUserToWhitelist(address).encodeABI();
+  const kycContractInstance = await waitForContract(KYC_CONTRACT_NAME);
+  return kycContractInstance.methods.addUserToWhitelist(address).encodeABI();
 }
 
 export async function removeUserFromWhitelist(address) {
-  const shieldContractInstance = await waitForContract(SHIELD_CONTRACT_NAME);
-  return shieldContractInstance.methods.removeUserFromWhitelist(address).encodeABI();
+  const kycContractInstance = await waitForContract(KYC_CONTRACT_NAME);
+  return kycContractInstance.methods.removeUserFromWhitelist(address).encodeABI();
+}
+
+export async function validateCertificate(certificate, ethereumAddressSignature) {
+  const kycContractInstance = await waitForContract(KYC_CONTRACT_NAME);
```

Comment:
No,  if it's null then the call will throw as expected. There's no need to check that.  The whole point of the call is to validate the certificate, it will throw if the certificate is invalid, for whatever reason.